lib/logger/utils.ts

```diff
@@ -322,11 +322,15 @@ export function validateLogLevel(
 }
 
 // Can't use `util/regex` because of circular reference to logger
-const urlRe = /[a-z]{3,9}:\/\/[^@/]+@[a-z0-9.-]+/gi;
+const urlRe = /(?:[a-z]{3,9}:\/\/[^@/]+@[a-z0-9.-]+|data:[^,]+,[^"]*)/gi;
 const urlCredRe = /\/\/[^@]+@/g;
+const dataUriCredRe = /data:[^,]+,.*$/g;
 
 export function sanitizeUrls(text: string): string {
   return text.replace(urlRe, (url) => {
+    if (url.startsWith('data:')) {
+      return url.replace(dataUriCredRe, 'data:**redacted**');
```

Comment:
Does all data really need replacing?

Reply:
IMO, yes cause the part after data may contain sensitive information such as the issuer of the uri along with the key itself. This is what we try to follow in our firm as well.